python/mxnet/gluon/block.py

```diff
@@ -1024,6 +1024,11 @@ def imports(symbol_file, input_names, param_file=None, ctx=None):
             ret.collect_params().load(param_file, ctx=ctx)
         return ret
 
+    def __repr__(self):
+        s = '{name}(\n{modstr}\n)'
+        modstr = '\n'.join(['{block}'.format(block=self.__dict__['_cached_graph'][-1])])
```

Comment:
why use `__dict__`? or -1 index?

Reply:
would it be more useful to show also the input count and output count?

Reply:
corrected __dict__ to self._cached_graph. 
-1 index as the last element of cached_graph is the output. I could have used cached_graph[1], but did not want to hit index out of bounds (in case it ever happens).

Reply:
I think showing all layers like how HybridSequential prints, would be most useful, but not sure how to do that for Symbol API and SymbolBlock models.

Reply:
Printing the inner structure might be too complicated and also not readable, thus I suggested printing the count.

Why would `cached_graph[1]` ever hit index out of bounds?

Reply:
something like this?

```
SymbolBlock(
<Symbol dense1_fwd>
)
Number of inputs: 1
Number of outputs: 1
```

Reply:
`<Symbol dense1_fwd>: 1->1` ?

Reply:
@szha thanks for the inputs. submitted after addressing the comments. Added an example of multiple outputs in the PR description